backend/src/LibExecution/Interpreter.fs

```diff
@@ -185,8 +185,14 @@ let execute (exeState : ExecutionState) (vm : VMState) : Ply<Dval> =
 
         | Lambda(parentContext, lambdaID) ->
           let lambda =
-            Map.findUnsafe (parentContext, lambdaID) vm.lambdaInstrCache
+            (match Map.tryFind (parentContext, lambdaID) vm.lambdaInstrCache with
+             | Some l -> l
+             | None ->
+               match Map.tryFind (Source, lambdaID) vm.lambdaInstrCache with
+               | Some l -> l
+               | None -> raiseRTE (RTE.VariableNotFound "lambda not found")) // TODO better error
```

Comment:
Hm, why do we need this fallback case? (`Source`) - When is that code hit?

Reply:
I suppose this is for the cases that we were trying to fix for -- but will the solution work if there are multiple nested lambdas? Or if a lambda is returned by a fn and used later on?

Reply:
This is hit when running functions that take a lambda as an argument (e.g. List.takeWhile, List.sortByComparator, List.sortBy, List.member, List.map, etc.)

The lambda couldn't be found in the `currentFrame.context` which is a `PackageFn`, instead it is found in the `Source`, I wasn't sure if it was the best solution.

As for `will the solution work if there are multiple nested lambdas` not sure yet, as I am encountering other errors in these cases that don't seem to be related to this issue